src/components/uploadPicture/uploadImage.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useEffect, useCallback, useRef } from "react";
+import styles from "./uploadImage.module.css";
+import { trpc } from "@/client/lib/trpc";
+import axios from "axios";
+import CameraIcon from "@mui/icons-material/PhotoCamera";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import CircularProgress from "@mui/material/CircularProgress";
+import Typography from "@mui/material/Typography";
+
+interface UploadImageComponentProps {
+  title: string;
+  fileName: string;
+  onUpload: (fileId: string) => void;
+}
+
+enum UploadStep {
+  Start,
+  TakePicture,
+  UploadPicture,
+}
+
+const UploadImageComponent: React.FC<UploadImageComponentProps> = ({
+  title,
+  fileName,
+  onUpload,
+}) => {
+  const [currentStep, setCurrentStep] = useState(UploadStep.Start);
+  const [uploading, setUploading] = useState(false);
+  const [uploadSuccess, setUploadSuccess] = useState(false);
+  const [capturedImage, setCapturedImage] = useState<Blob | null>(null);
+
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
```

Comment:
🐧 I think typing this as
```suggestion
  const fileInputRef = useRef<HTMLInputElement>(null);
```
works too (slightly cleaner)